packages/material-ui/src/IconButton/IconButton.test.js

```diff
@@ -85,6 +85,24 @@ describe('<IconButton />', () => {
     assert.strictEqual(wrapper.props().centerRipple, true);
   });
 
+  it('should render with sizeSmall class when prop size="small" provided', () => {
+    const wrapper = shallow(<IconButton size="small">book</IconButton>);
```

Comment:
Can we use the mount API? We want to kill all the shallow tests cc @eps1lon

Reply:
Agreed. It's not actually testing if that style will reach the browser. 

1. For static behavior this is mostly fine since we have regression tests but for everything else we still need these unit tests.
2. It breaks if we change what components we use inside render even if those components do not affect the DOM

Reply:
> We want to kill all the shallow tests

Oh boy, how time flies. :sweat_smile: I can remember us preferring shallow tests for their speed when implementing the table pagination. I'll update the tests.

@eps1lon It tests if the proper CSS classes are applied. It obviously can't test if it looks right (that's what the regression tests are there for).

Reply:
> @eps1lon It tests if the proper CSS classes are applied. It obviously can't test if it looks right (that's what the regression tests are there for).

Sure. What I meant was that the proper semantics are applied to the DOM elements.